druntime/test/shared/src/utils.di

```diff
@@ -1,12 +1,29 @@
 module utils;
 
+version (OSX)
+    version = Darwin;
+else version (iOS)
+    version = Darwin;
+else version (TVOS)
+    version = Darwin;
+else version (WatchOS)
+    version = Darwin;
+
 version (Windows)
     enum dllExt = "dll";
-else version (darwin)
+else version (Darwin)
     enum dllExt = "dylib";
 else
     enum dllExt = "so";
 
+// on some platforms, dlclose() is a no-op
+version (Darwin)
+    enum isDlcloseNoop = true; // since macOS ~10.12.6 if shared lib uses TLS: https://github.com/rust-lang/rust/issues/28794#issuecomment-368693049
```

Comment:
This hasn't been a problem for DMD - simply because it doesn't support shared druntime/Phobos on Darwin, and these tests are skipped. This matters for LDC though, which has tweaked these tests.